### PR TITLE
Fixed when roost removes flying type

### PR DESCRIPTION
**Pull request by IshanP02**
Roost was removing flying type on the next turn instead of during the current turn. Just added a condition called GROUNDED that does the same thing as IGNORE_FLYING, but on the current turn instead of the next turn.

### CODE REVIEW

**Review by Flashfyre — CHANGES_REQUESTED** [review]
There are plenty of places already that use IGNORE_FLYING, and don't know about GROUNDED, can you ensure those still work properly?

**Comment by IshanP02** [comment]
I did not change anything that used IGNORE_FLYING or IGNORE_FLYING itself aside from roost, and I can confirm that roost works properly through local testing.

**Comment by IshanP02** [comment]
I might've misunderstood, are you asking if, for example, a Pokemon uses smack down, and then on the next turn, the pokemon hit uses roost, so both IGNORE_FLYING and GROUNDED are on the pokemon at the same time, does this cause issues. If so, looking at the code, all IGNORE_FLYING and GROUNDED do are remove the flying type of a Pokemon if it has it, so it shouldn't cause any issues, whichever is called first will just remove the flying type and the next will do nothing because there is no flying type to remove.

**Comment by Flashfyre** [comment]
I was just asking you to look up every case of IGNORE_FLYING and check if it should also be applied to GROUNDED.

**Comment by IshanP02** [comment]
Understood, I just looked through and added a GROUNDED check to levitate. That should cover all instances of IGNORE_FLYING that need to be applied to GROUNDED.

**Review by Flashfyre — APPROVED** [review]

File: src/field/pokemon.ts [hunk 1]
```diff
@@ -698,6 +698,12 @@ export default abstract class Pokemon extends Phaser.GameObjects.Container {
         types.splice(flyingIndex, 1);
     }
 
+    if (forDefend && (this.getTag(BattlerTagType.GROUNDED) || this.scene.arena.getTag(ArenaTagType.GRAVITY))) {
```
Comment (Flashfyre) [inline src/field/pokemon.ts:701]: One more thing, why are we duplicating the previous block here but with grounded? Is this necessary, can they not be combined?

**Review by Flashfyre — CHANGES_REQUESTED** [review]

File: src/field/pokemon.ts (hunk 1 above)
Comment (IshanP02) [reply src/field/pokemon.ts:701]: Yeah they could be combined, not sure why I duplicated it to be honest. Same result either way, but would be cleaner combined yes, I'll do that right now.

Comment (IshanP02) [reply src/field/pokemon.ts]: Done

**Comment by IshanP02** [comment]
Just posting this comments because this pull request still says changes requested instead of review required and I think posting this comment will change that?

**Pull request merged**